packages/web-client/app/components/card-pay/restorable-workflow-component/index.ts

```diff
@@ -0,0 +1,41 @@
+import { Workflow } from '@cardstack/web-client/models/workflow';
+import WorkflowPersistence from '@cardstack/web-client/services/workflow-persistence';
+import { getOwner } from '@ember/application';
+import RouterService from '@ember/routing/router-service';
+import Component from '@glimmer/component';
+import { cached, tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+
+interface WorkflowConstructor<TWorkflow> {
+  new (owner?: any, workflowPersistenceId?: string): TWorkflow;
+}
+
+export default abstract class RestorableWorkflowComponent<
+  TWorkflow extends Workflow
+> extends Component {
+  @service declare workflowPersistence: WorkflowPersistence;
+  @service declare router: RouterService;
+  @tracked isInitializing = true;
+
+  get workflowPersistenceId() {
+    return this.router.currentRoute.queryParams['flow-id']!;
+  }
+
+  abstract get workflowClass(): WorkflowConstructor<TWorkflow>;
```

Comment:
Just an observation:

There's an interesting (Ember?) bug here, I think? In normal typescript we shouldn't need to declare that an abstract property is a getter for it to work as a getter in components that extend this class, but here the subclasses throw errors if this is replaced with `abstract workflowClass: WorkflowConstructor<TWorkflow>`. I guess extending glimmer's `Component` has some side effects. 🤔

Reply:
Yes, I wrestled with this quite a bit yesterday. I don't think it is Ember-related, but could be wrong.